pd-control: add new config of balance-hot-region-scheduler (#6148)
pingcap/docs: pd-control.md

{
  "min-hot-byte-rate": 100,
  "min-hot-key-rate": 10,
  "min-hot-query-rate": 10,
  "max-zombie-rounds": 3,
  "max-peer-number": 1000,
  "byte-rate-rank-step-ratio": 0.05,
  "key-rate-rank-step-ratio": 0.05,
  "query-rate-rank-step-ratio": 0.05,
  "count-rank-step-ratio": 0.01,
  "great-dec-ratio": 0.95,
  "minor-dec-ratio": 0.99,
  "src-tolerance-ratio": 1.05,
  "dst-tolerance-ratio": 1.05,
  "read-priorities": [
    "query",
    "byte"
  ],
  "write-leader-priorities": [
    "key",
    "byte"
  ],
  "write-peer-priorities": [
    "byte",
    "key"
  ],
  "strict-picking-store": "true",
  "enable-for-tiflash": "true"
}
```

> - `min-hot-byte-rate` means the smallest number byte counted, which is usually 100.

Reviewer: ```suggestion
- `min-hot-byte-rate` means the smallest number of bytes to be counted, which is usually 100.
```

    ```bash
    >> scheduler config balance-hot-region-scheduler set min-hot-byte-rate 100
    ```

> - `min-hot-key-rate` means the smallest number key counted, which is usually 10.

Reviewer: ditto

    ```bash
    >> scheduler config balance-hot-region-scheduler set min-hot-key-rate 10
    ```

> - `min-hot-query-rate` means the smallest number query counted, which is usually 10.

Reviewer: ditto

Author: fixed in https://github.com/pingcap/docs/pull/6148/commits/053c5c22da4e5132da7687fed951d12cb76a54b5.

- `src-tolerance-ratio` and `dst-tolerance-ratio` are configuration items for the expectation scheduler. The smaller the `tolerance-ratio`, the easier it is for scheduling. When redundant scheduling occurs, you can appropriately increase this value.

    ```bash
       >> scheduler config balance-hot-region-scheduler set src-tolerance-ratio 1.1
    ```

- `read-priorities`,`write-leader-priorities`, and `write-peer-priorities` control the first and second dimensions which have the first priority of balance when different types of hotspots are delt with. For hotspots of  `read` and `write-leader` types, the available dimensions are `query`, `byte`, and `key`. For hotspots of `write-peer`, the available dimensions are `byte` and `key`. If not all the cluster components are upgraded to v5.2 and later, these configurations will not take effect, and the compatible configurations will be used. Usually, you do not need to modify these configuration items.

    ```bash
    >> scheduler config balance-hot-region-scheduler set read-priorities query,byte
    ```

> - `strict-picking-store` is a switch that controls the search space of hotspot scheduling. When it is enabled and if stability is ensured, hotspot scheduling will start. Usually it is enabled. When it is disabled, only the balance of dimensions of the first priority is enusred, which might reduce the balance of other dimensions. Usually, you do not need to modify this configuration item.

Reviewer: `When it is enabled and if stability is ensured, hotspot scheduling will start` confuse me

> - `strict-picking-store` is a switch that controls the search space of hot region scheduling. When it is enabled and if stability is ensured, hot region scheduling will start. Usually it is enabled. When it is disabled, only the balance of dimensions of the first priority is enusred, which might reduce the balance of other dimensions. Usually, you do not need to modify this configuration item.

Author: ”打开时会在保证稳定性的前提下进行热点调度“，PTAL, thanks~ @lhy1024 
```suggestion
- `strict-picking-store` is a switch that controls the search space of hot region scheduling. When it is enabled, hot region scheduling starts to perform on the premise of stability. Usually it is enabled. When it is disabled, only the balance of dimensions of the first priority is enusred, which might reduce the balance of other dimensions. Usually, you do not need to modify this configuration item.
```

Reviewer: Maybe the original description is just not accurate, I will correct it in another pr.

Reviewer: https://github.com/pingcap/docs-cn/pull/6888/files

- `src-tolerance-ratio` and `dst-tolerance-ratio` are configuration items for the expectation scheduler. The smaller the `tolerance-ratio`, the easier it is for scheduling. When redundant scheduling occurs, you can appropriately increase this value.

    ```bash

>        >> scheduler config balance-hot-region-scheduler set src-tolerance-ratio 1.1

Reviewer 2: ```suggestion
>> scheduler config balance-hot-region-scheduler set src-tolerance-ratio 1.1
```

>        >> scheduler config balance-hot-region-scheduler set src-tolerance-ratio 1.1

Reviewer 2: ```suggestion
    >> scheduler config balance-hot-region-scheduler set src-tolerance-ratio 1.1
```

    ```

> - `read-priorities`,`write-leader-priorities`, and `write-peer-priorities` control which dimensions have the priority for hot Region scheduling and support two dimensions.

Reviewer 2: ```suggestion
- `read-priorities`, `write-leader-priorities`, and `write-peer-priorities` control which dimension the scheduler prioritizes for hot Region scheduling. Two dimensions are supported for configuration.
```

>     - `read-priorities` and `write-leader-priorities` control the dimensions to be prioritized for read and write-leader hot Region scheduling. The available dimensions are `query`, `byte` and `key`.

Reviewer 2: ```suggestion
    - `read-priorities` and `write-leader-priorities` control which dimensions the scheduler prioritizes for scheduling hot Regions of the read and write-leader types. The dimension options are `query`, `byte`, and `key`.
```

>     - `write-peer-priorities` controls the dimension to be prioritized for write-peer hot Region scheduling. The available dimensions are `byte` and `key`.

Reviewer 2: ```suggestion
    - `write-peer-priorities` controls which dimensions the scheduler prioritizes for scheduling hot Regions of the write-peer type. The dimension options are `byte` and `key`.
```

    
    > **Note:**
    >

>     > If not all the cluster components are upgraded to v5.2 and later, the configuration of `query` dimension does not take effect. If some components are upgraded, the `byte` and `key` dimensions still by default have the priority for hot Region scheduling. If all the components of the cluster are upgraded to v5.2 and later, such a compatible configuration still takes effect. You can view the real-time configuration using `pd-ctl` command. Usually, you do not need to modify these configurations.

Reviewer: ```suggestion
    > If not all the cluster components are upgraded to v5.2 and later, the configuration of `query` dimension does not take effect. If some components are upgraded, the `byte` and `key` dimensions still by default have the priority for hot Region scheduling. After all the components of the cluster are upgraded to v5.2 and later, such a compatible configuration still takes effect. You can view the real-time configuration using `pd-ctl` command. Usually, you do not need to modify these configurations.
```

>     > If not all the cluster components are upgraded to v5.2 and later, the configuration of `query` dimension does not take effect. If some components are upgraded, the `byte` and `key` dimensions still by default have the priority for hot Region scheduling. After all the components of the cluster are upgraded to v5.2 and later, such a compatible configuration still takes effect. You can view the real-time configuration using `pd-ctl` command. Usually, you do not need to modify these configurations.

Reviewer 2: ```suggestion
    > If a cluster component is earlier than v5.2, the configuration of `query` dimension does not take effect. If some components are upgraded to v5.2 or later, the `byte` and `key` dimensions still by default have the priority for hot Region scheduling. After all components of the cluster are upgraded to v5.2 or later, such a configuration still takes effect for compatibility. You can view the real-time configuration using the `pd-ctl` command. Usually, you do not need to modify these configurations.
```

    ```bash
    >> scheduler config balance-hot-region-scheduler set read-priorities query,byte
    ```

> - `strict-picking-store` controls the search space of hot Region scheduling. Usually it is enabled. When it is enabled, hot Region scheduling ensures the balance between the two configured dimensions. When it is disabled, hot Region scheduling only ensures the balance of dimensions with the first priority, which might reduce the balance of other dimensions. Usually, you do not need to modify this configuration.

Reviewer 2: ```suggestion
- `strict-picking-store` controls the search space of hot Region scheduling. When it is enabled, the scheduler performs hot Region scheduling while ensuring stability. Usually, it is enabled. When it is disabled, hot Region scheduling only ensures the balance of dimensions with the first priority, which might reduce the Region balance of other dimensions. Usually, you do not need to modify this configuration.
```

Author: In https://github.com/pingcap/docs-cn/pull/6888, the Chinese in this section has been improved because the author thinks `ensuring stability` makes readers confusing. PTAL~

Reviewer 2: Thanks! I have modified my comment. PTAL again.

> - `strict-picking-store` controls the search space of hot Region scheduling. Usually it is enabled. When it is enabled, hot Region scheduling ensures the balance between the two configured dimensions. When it is disabled, hot Region scheduling only ensures the balance of dimensions with the first priority, which might reduce the balance of other dimensions. Usually, you do not need to modify this configuration item.

Author: ```suggestion
- `strict-picking-store` controls the search space of hot Region scheduling. Usually it is enabled. When it is enabled, hot Region scheduling ensures the balance between the two configured dimensions. When it is disabled, hot Region scheduling only ensures the balance of dimensions with the first priority, which might reduce the balance of other dimensions. Usually, you do not need to modify this configuration.
```

    ```bash
    >> scheduler config balance-hot-region-scheduler set strict-picking-store true
    ```

> - `enable-for-tiflash` controls whether hot Region scheduling takes effect for TiFlash. Usually it is enabled. When it is disabled, the hot Region scheduling between TiFlash instances do not start.

Reviewer: ```suggestion
- `enable-for-tiflash` controls whether hot Region scheduling takes effect for TiFlash instances. Usually it is enabled. When it is disabled, the hot Region scheduling between TiFlash instances do not start.
```

> - `enable-for-tiflash` controls whether hot Region scheduling takes effect for TiFlash instances. Usually it is enabled. When it is disabled, the hot Region scheduling between TiFlash instances do not start.

Reviewer 2: ```suggestion
- `enable-for-tiflash` controls whether hot Region scheduling takes effect for TiFlash instances. Usually, it is enabled. When it is disabled, the hot Region scheduling between TiFlash instances is not performed..
```
